core/document/src/layers/simple_shape.rs

```diff
@@ -0,0 +1,145 @@
+use glam::DAffine2;
+use glam::DMat2;
+use glam::DVec2;
+
+use kurbo::Affine;
+use kurbo::Shape as KurboShape;
+
+use crate::intersection::intersect_quad_bez_path;
+use crate::LayerId;
+use crate::Quad;
+use kurbo::BezPath;
+
+use super::style;
+use super::style::PathStyle;
+use super::LayerData;
+
+use serde::{Deserialize, Serialize};
+use std::fmt::Write;
+
+fn glam_to_kurbo(transform: DAffine2) -> Affine {
+	Affine::new(transform.to_cols_array())
+}
+
+#[derive(Debug, Clone, PartialEq, Deserialize, Serialize)]
+pub struct Shape {
+	pub path: BezPath,
+	pub style: style::PathStyle,
+	pub render_index: i32,
+	pub solid: bool,
+}
+
+impl LayerData for Shape {
+	fn render(&mut self, svg: &mut String, transforms: &mut Vec<DAffine2>) {
+		let mut path = self.path.clone();
+		let transform = self.transform(transforms);
+		let inverse = transform.inverse();
+		if !inverse.is_finite() {
+			let _ = write!(svg, "<!-- Svg shape has an invalid transform -->");
+			return;
+		}
+		path.apply_affine(glam_to_kurbo(transform));
+
+		let _ = writeln!(svg, r#"<g transform="matrix("#);
+		inverse.to_cols_array().iter().enumerate().for_each(|(i, f)| {
+			let _ = svg.write_str(&(f.to_string() + if i != 5 { "," } else { "" }));
+		});
+		let _ = svg.write_str(r#")">"#);
+		let _ = write!(svg, r#"<path d="{}" {} />"#, path.to_svg(), self.style.render());
+		let _ = svg.write_str("</g>");
+	}
+
+	fn bounding_box(&self, transform: glam::DAffine2) -> Option<[DVec2; 2]> {
+		let mut path = self.path.clone();
+		if transform.matrix2 == DMat2::ZERO {
+			return None;
+		}
+		path.apply_affine(glam_to_kurbo(transform));
+
+		use kurbo::Shape;
+		let kurbo::Rect { x0, y0, x1, y1 } = path.bounding_box();
+		Some([(x0, y0).into(), (x1, y1).into()])
+	}
+
+	fn intersects_quad(&self, quad: Quad, path: &mut Vec<LayerId>, intersections: &mut Vec<Vec<LayerId>>) {
+		if intersect_quad_bez_path(quad, &self.path, self.solid) {
+			intersections.push(path.clone());
+		}
+	}
+}
+
+impl Shape {
+	pub fn transform(&self, transforms: &[DAffine2]) -> DAffine2 {
+		let start = match self.render_index {
+			-1 => 0,
+			x => (transforms.len() as i32 - x).max(0) as usize,
+		};
+		transforms.iter().skip(start).cloned().reduce(|a, b| a * b).unwrap_or(DAffine2::IDENTITY)
+	}
+
+	pub fn shape(sides: u8, style: PathStyle) -> Self {
+		use std::f64::consts::PI;
+		fn unit_rotation(theta: f64) -> DVec2 {
+			DVec2::new(theta.sin(), theta.cos())
+		}
+		let mut path = kurbo::BezPath::new();
+		let apothem_offset_angle = 2. * PI / (sides as f64);
+		let offset = ((sides + 1) % 2) as f64 * PI / 2.; // rotate odd sided shaps by 90 degrees
```

Comment:
Parentheses around `((sides + 1) % 2) as f64 * PI` would be nice, to be clear about the order of operations for the division.